generic-map.js

```diff
@@ -159,6 +160,14 @@ GenericMap.prototype.entries = function () {
     });
 };
 
+GenericMap.prototype.iterate = function() {
+    // v.1 iterator
+    var self = this;
+    return new Iterator(function() {
+        return self.iterator.next().value;
```

Comment:
Observing the whole code, I don't like the behaviour of the .iterator() method of other types of collections. I have just added this method as a compatibility issue. As you can see, I am even not sure, whether the old-style iterator should return value, or the key, or some kind of pair (value,key). All variants are bad IMHO. The v2 iterator behaviour is much better.